cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORGenerator.java

```diff
@@ -1452,20 +1425,43 @@ private final int _encode2(int i, int outputPtr, String str, int len,
             }
             // Yup, a surrogate pair
             if (c > SURR1_LAST) { // must be from first range; second won't do
-                _throwIllegalSurrogate(c);
+                if (_cfgLenientUnicodeEncoding) {
```

Comment:
One small suggestion: change `_throwIllegalSurrogate` to contain this check, and either throw exception or return surrogate value. That way rarely needed logic contained in that method, bit more compact on call site.

Reply:
Yes good idea, done.